stable/dex/values.yaml

```diff
@@ -3,14 +3,16 @@
 # Declare name/value pairs to be passed into your templates.
 # name: value
 
-image: quay.io/dexidp/dex
-imageTag: "v2.17.0"
+image: mesosphere/dex
```

Comment:
We are setting these values in `Kuberenetes-base-addons`. Should we continue to do that? It helps keep the chart close to the upstream and also it is easier to control these settings from `kubernetes-base-addons`

Reply:
See my above comment

Reply:
Maybe for consistency and to avoid any confusion we just follow the style of keeping the chart close to upstream version and modifying it in `kubernetes-base-addons`. The flip side of it is that this diff should always be carry over all subsequent bumps. I also want to avoid the case where somebody mistakenly updates the image here but not in `kubernetes-base-addons`.

Reply:
@GoelDeepak I would remove the image repo/tag in the kubernetes-base-addons for dex so that there's only one place we set the image repo/tag so there will be no confusion where to bump the image.

Reply:
Sorry @jieyu for pushing on this. I ideally prefer there than here just because every time you do a dex bump you would have to remember to port over this change. It is much better to not have to do that and be taken care automatically in `kubernetes-base-addons`. One less thing to worry about. Is there any benefit of keeping it here?

Reply:
hum, ok, that's a fair point. You're saying that if do in addon config, in many case, we don't need to bump the chart.